File: testresources/java/accuracy/src/domain/indirect/BaseIndirect.java
package domain.indirect;

import technology.direct.subclass.CallInstanceSubClassDOA;
import technology.direct.subclass.CallInstanceSubSubClassDOA;
import domain.indirect.indirectto.POI;

public class BaseIndirect {
	
	protected CallInstanceSubClassDOA subDao;
	protected CallInstanceSubSubClassDOA subSubDao;
	

	public BaseIndirect(){
		
	}

	public CallInstanceSubClassDOA getProfileInformation(int i, String s){
		return subDao;
	}

	public POI getProfileInformation(String s, int i){
		return new POI();
	}

	public POI getMethodCorrectlyByDerivedArgumentType(String s, String i, int j){
		return new POI();
	}

	public CallInstanceSubClassDOA getMethodCorrectlyByDerivedArgumentType(String s, int i, int j){
		return subDao;
	}

	public CallInstanceSubClassDOA getMethodCorrectlyByHeuristic(int i, String s, int j){
		return subDao;
	}

	public POI getMethodCorrectlyByHeuristic(String s, String p, int i){
		return new POI();
	}

}

File: testresources/csharp/benchmark/Domain/Indirect/BaseIndirect.cs
using Technology.Direct.Subclass;
using Domain.Indirect.IndirectTo;

namespace Domain.Indirect
{
	public class BaseIndirect
	{

        		protected CallInstanceSubClassDOA subDao;

        		protected CallInstanceSubSubClassDOA subSubDao;

		public BaseIndirect()
		{
		}

	public CallInstanceSubClassDOA getProfileInformation(int i, string s){
		return subDao;
	}

	public POI getProfileInformation(string s, int i){
		return new POI();
	}

	public POI getMethodCorrectlyByDerivedArgumentType(string s, string j, int j){
		return new POI();
	}

	public CallInstanceSubClassDOA getMethodCorrectlyByDerivedArgumentType(string s, int i, int j){
		return subDao;
	}

	public CallInstanceSubClassDOA getMethodCorrectlyByHeuristic(int i, string s, int j){
		return subDao;
	}

	public POI getMethodCorrectlyByHeuristic(string s, string p, int i){
		return new POI();
	}

	}
}
